app/javascript/components/Show.js

```diff
@@ -1,33 +1,86 @@
-import React, { useState, useEffect } from "react"
+import React from 'react';
+import { Steps, Button, message } from 'antd';
+import 'antd/dist/antd.css';
 
-function Show(id){
-  useEffect(() => {
-    fetchExam();
-  }, []);
-
-  const [exam, setExam] = useState({});
-
-  const fetchExam = async () => {
-    const fetchExam = await fetch('/api/v1/exams/' + id.id);
-    const exam = await fetchExam.json();
-    setExam(exam);
-  };
-  var rows = [];
-
-  if (exam.length) {
-    rows = exam.map((item, index) => (
-      <div key={index}>
-        Question: {item.question_content}
-        Level: {item.level}
-      </div>
-  ))
- }
-
-  return (
-    <div>
-      {rows}
-    </div>
-  )
+class Show extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      exam: [],
+      isLoaded: false,
+      current: 0,
+    }
+  }
+  next() {
+    var current = this.state.current + 1;
+    this.setState({current});
+  }
+  prev() {
+    var current = this.state.current - 1;
+    this.setState({current});
+  }
+  componentDidMount() {
+    var path = window.location.pathname
+    var exam_id = path.substring(path.lastIndexOf('/') + 1)
+    fetch(`/api/v1/exams/` + exam_id)
+      .then(res => res.json())
+      .then(json => {
+        this.setState({
+          isLoaded: true,
+          exam: json
+        })
+      });
+  }
+  render() {
+    var { isLoaded, exam } = this.state;
+    var { current } = this.state;
```

Comment:
1. nên dùng const 
2. có thể viết gọn lại:
const {  isLoaded, exam, current, } = this.state;

Reply:
Update